<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MatrixWindow</class>
 <widget class="QMainWindow" name="MatrixWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1036</width>
    <height>500</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>500</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>100000</width>
    <height>10000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Матрица</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #171b22;
color: #fff;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>600</width>
        <height>100</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>100000</width>
        <height>1000000</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
      </property>
      <property name="title">
       <string>Матрица</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <property name="spacing">
          <number>612</number>
         </property>
         <property name="leftMargin">
          <number>12</number>
         </property>
         <property name="topMargin">
          <number>12</number>
         </property>
         <property name="rightMargin">
          <number>12</number>
         </property>
         <property name="bottomMargin">
          <number>12</number>
         </property>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Preferred</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>10</width>
             <height>10</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>12</number>
           </property>
           <property name="leftMargin">
            <number>12</number>
           </property>
           <property name="topMargin">
            <number>12</number>
           </property>
           <property name="rightMargin">
            <number>12</number>
           </property>
           <property name="bottomMargin">
            <number>12</number>
           </property>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTableWidget" name="matrixTable">
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="styleSheet">
              <string notr="true">QTableWidget {
    background-color: transparent;
    alternate-background-color: #171b22;
    show-decoration-selected: 0;
    border: 0px;
    border-radius: 5px;
}
QTableWidget::item {
	background-color: #7289DA;
    border-radius: 5px;
}</string>
             </property>
             <property name="inputMethodHints">
              <set>Qt::ImhDigitsOnly</set>
             </property>
             <property name="frameShape">
              <enum>QFrame::WinPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Plain</enum>
             </property>
             <property name="lineWidth">
              <number>1</number>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <property name="showGrid">
              <bool>true</bool>
             </property>
             <property name="gridStyle">
              <enum>Qt::DashLine</enum>
             </property>
             <property name="rowCount">
              <number>0</number>
             </property>
             <property name="columnCount">
              <number>0</number>
             </property>
             <attribute name="horizontalHeaderVisible">
              <bool>false</bool>
             </attribute>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_5">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Preferred</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>10</width>
             <height>10</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPushButton" name="clearDataButton">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QPushButton {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}</string>
           </property>
           <property name="text">
            <string>Очистить матрицу</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>100</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>1000000</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
      </property>
      <property name="title">
       <string>Обработка матрицы</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>12</number>
         </property>
         <property name="leftMargin">
          <number>12</number>
         </property>
         <property name="topMargin">
          <number>12</number>
         </property>
         <property name="rightMargin">
          <number>12</number>
         </property>
         <property name="bottomMargin">
          <number>12</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Размер матрицы</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="xSizeMatrix">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
           </property>
           <property name="maximum">
            <number>9999999</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="minimumSize">
            <size>
             <width>10</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>10</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">	background-color: transparent;</string>
           </property>
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="ySizeMatrix">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
           </property>
           <property name="maximum">
            <number>9999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="createMatrixTable">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}</string>
         </property>
         <property name="text">
          <string>Создать матрицу</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>12</number>
         </property>
         <property name="leftMargin">
          <number>12</number>
         </property>
         <property name="topMargin">
          <number>12</number>
         </property>
         <property name="rightMargin">
          <number>12</number>
         </property>
         <property name="bottomMargin">
          <number>12</number>
         </property>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Заполнить матрицу числами от</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="minIntMatrix">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::PlusMinus</enum>
           </property>
           <property name="minimum">
            <number>-99999999</number>
           </property>
           <property name="maximum">
            <number>9999999</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>до</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="maxIntMatrix">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
           </property>
           <property name="minimum">
            <number>-99999999</number>
           </property>
           <property name="maximum">
            <number>9999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="setMatrixData">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}</string>
         </property>
         <property name="text">
          <string>Заполнить матрицу</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>12</number>
         </property>
         <property name="leftMargin">
          <number>12</number>
         </property>
         <property name="topMargin">
          <number>12</number>
         </property>
         <property name="rightMargin">
          <number>12</number>
         </property>
         <property name="bottomMargin">
          <number>12</number>
         </property>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="productOfNegativeNumbers">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>Произведение отрицательных чётных чисел:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="summElements">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>Сумма элементов на главной диагонали: </string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLineEdit" name="answerArray">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
           </property>
           <property name="placeholderText">
            <string>Одномерный массив: сумма чётных чисел каждой строки</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0">
           <property name="spacing">
            <number>12</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QPushButton" name="startProcessBtn">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}</string>
             </property>
             <property name="text">
              <string>Обработать</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="saveDataToFileBtn">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}</string>
             </property>
             <property name="text">
              <string>Записать итог в файл</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
